File: Lab4CrossingWords/Lab4CrossingWords/CrossPointInfo.cs
using System;

namespace Lab4CrossingWords
{
    public class CrossPointInfo
    {
        public int Word1Number { get; set; }
        public int Word2Number { get; set; }

        public char Letter { get; set; }

        public int W1Pos { get; set; }
        public int W2Pos { get; set; }

        public bool IsSameWordsCross(CrossPointInfo toCheck)
        {
            return (Word1Number == toCheck.Word1Number && Word2Number == toCheck.Word2Number);
        }

        public override string ToString()
        {
            return $"{Word1Number} {Word2Number}";
        }


        public int GetThisCross(int firstKey)
        {
            if (firstKey == Word1Number)
            {
                return W1Pos;
            }

            if (firstKey == Word2Number)
            {
                return W2Pos;
            }

            throw new ArgumentException();
        }


        public int GetPairedCrossNum(int oneOfPairNum)
        {
            if (oneOfPairNum == Word1Number)
            {
                return Word2Number;
            }

            if (oneOfPairNum == Word2Number)
            {
                return Word1Number;
            }

            throw new ArgumentException();
        }

    }
}

File: Lab4CrossingWords/TestL4/CrossPointInfo.cpp
#include "stdafx.h"

#include "CrossPointInfo.h"

CrossPointInfo::CrossPointInfo(int w1n, int w2n, char l, int w1pos, int w2pos)
{
	W1Pos = w1pos;
	W2Pos = w2pos;
	Letter = l;
	Word1Number = w1n;
	Word2Number = w2n;
}

bool CrossPointInfo::IsSameWordsCross(CrossPointInfo toCheck)
{
	return (Word1Number == toCheck.Word1Number && Word2Number == toCheck.Word2Number);
}

std::string CrossPointInfo::ToString()
{
	return "Not implimented"; //TODO: add this   // Word1Number.
}

int CrossPointInfo::GetThisCross(int firstKey)
{
	if (firstKey == Word1Number)
	{
		return W1Pos;
	}

	if (firstKey == Word2Number)
	{
		return W2Pos;
	}
}

int CrossPointInfo::GetPairedCrossNum(int oneOfPairNum)
{
	if (oneOfPairNum == Word1Number)
	{
		return Word2Number;
	}

	if (oneOfPairNum == Word2Number)
	{
		return Word1Number;
	}
}


